lib/elsa/group/worker.ex

```diff
@@ -77,12 +78,19 @@ defmodule Elsa.Group.Worker do
   end
 
   def handle_continue(:subscribe, state) do
-    case subscribe(state) do
-      {:ok, pid} ->
-        Process.monitor(pid)
-        {:noreply, %{state | subscriber_pid: pid}}
-
+    with :ok <- :brod.start_consumer(state.name, state.topic, state.config),
+         {:ok, pid} <- subscribe(state) do
+      Process.monitor(pid)
+      {:noreply, %{state | subscriber_pid: pid}}
+    else
       {:error, reason} ->
+        Logger.warn(
+          "Unable to subscribe to topic/partition/offset(#{state.topic}/#{state.partition}/#{state.offset}), reason #{
+            inspect(reason)
+          }"
+        )
+
+        Process.sleep(@start_failure_delay)
```

Comment:
I'd be curious to here why we're sleeping here. I'm guessing we're waiting for something else to stop that we can't easily hook into?

Reply:
Sleeping forces the client to wait before exit so it doesn’t immediately trigger the supervisor to self-destruct due to the restart limit when there’s a network issue beyond our control.

Reply:
Aaahhhh. JMS.